src/Integration.Vsix/CFamily/VcxProject/VcxRequestFactory.cs

```diff
@@ -53,11 +51,15 @@ internal class VcxRequestFactory : IRequestFactory
             ICFamilyRulesConfigProvider cFamilyRulesConfigProvider,
             ILogger logger,
             IThreadHandling threadHandling)
+// Suppress FP. The call to threadHandling.ThrowIfNotOnUIThread() inside the Lazy<> causes the
+// incorrect identify this constructor as needing to do a threading check.
+#pragma warning disable VSTHRD010 // Invoke single-threaded types on Main thread
```

Comment:
The code in this class was a bit more complicated, so it was simpler to just mark this as a false positive (which in this case it really is a false positive).